fix: irreplaceable turfs after crystall and non-icon floor in lavalend skin

Rerik007 opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
Из незаменяемых полов был лишь снег, теперь его можно откопать лопатой и удалить. + возможность снежка, также как и у снега после снегмашины.
У asteroid полов нет иконки после change turfa, чинит это для кристалла.
Также теперь, если вы будете убирать темы снега или джунглей -  вам вернётся старый турф. Стоит ли возвращать для остальных не знаю.
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
Закрывает багрепорт: https://discord.com/channels/617003227182792704/1078378165254238268
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

## Демонстрация изменений
![Снимок экрана 2023-03-02 114839](https://user-images.githubusercontent.com/120549107/222379778-47074c74-566a-46ca-8cad-35a20611a7d2.png)
![Снимок экрана 2023-03-04 135517](https://user-images.githubusercontent.com/120549107/222896185-012b9235-b0a5-41a2-b0da-049e961b196d.png)


<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->

Rerik007 committed:
fixess

Dimach commented:
Это решение после замены турфа лаваленда оставляет ВСЕГДА плейтинг, хотя далеко не факт что изначально там был именно он. Желательно переписать это так чтобы он запоминал какой конкретно турф там был до того

Rerik007 commented:
> Это решение после замены турфа лаваленда оставляет ВСЕГДА плейтинг, хотя далеко не факт что изначально там был именно он. Желательно переписать это так чтобы он запоминал какой конкретно турф там был до того

Хм, тоесть? Сейчас плитки лаваленда можно убрать только другими плитками.
У всех типов скинов подо всеми полами будет обычный платинг. Нужно чтобы под полом был старый пол? Или как-то изменить способ замены плиток лаваленда?
Раньше с лавалендом просто была проблема в этом:
![Снимок экрана 2023-03-04 092334](https://user-images.githubusercontent.com/120549107/222881580-2b6463e1-70a5-4ee6-88fd-a9f28cae67c8.png)

Rerik007 committed:
more best

Сделал чтобы иконки были базовые для нужного типа в данном случае.

Dimach commented:
> > Это решение после замены турфа лаваленда оставляет ВСЕГДА плейтинг, хотя далеко не факт что изначально там был именно он. Желательно переписать это так чтобы он запоминал какой конкретно турф там был до того
> 
> Хм, тоесть? Сейчас плитки лаваленда можно убрать только другими плитками. У всех типов скинов подо всеми полами будет обычный платинг. Нужно чтобы под полом был старый пол? Или как-то изменить способ замены плиток лаваленда? Раньше с лавалендом просто была проблема в этом: ![Снимок экрана 2023-03-04 092334](https://user-images.githubusercontent.com/120549107/222881580-2b6463e1-70a5-4ee6-88fd-a9f28cae67c8.png)

У меня например был стеклянный пол, на эту арею юзанули артефакт лаваленда и он заменился на снег, теперь мы убираем снег и получаем.. плейтинг, хотя изначально там был стеклянный пол. Можно ли сделать так чтобы эти лавалендовские турфы сохраняли что за пол там был раньше?

Rerik007 commented:
> > > Это решение после замены турфа лаваленда оставляет ВСЕГДА плейтинг, хотя далеко не факт что изначально там был именно он. Желательно переписать это так чтобы он запоминал какой конкретно турф там был до того
> > 
> > 
> > Хм, тоесть? Сейчас плитки лаваленда можно убрать только другими плитками. У всех типов скинов подо всеми полами будет обычный платинг. Нужно чтобы под полом был старый пол? Или как-то изменить способ замены плиток лаваленда? Раньше с лавалендом просто была проблема в этом: ![Снимок экрана 2023-03-04 092334](https://user-images.githubusercontent.com/120549107/222881580-2b6463e1-70a5-4ee6-88fd-a9f28cae67c8.png)
> 
> У меня например был стеклянный пол, на эту арею юзанули артефакт лаваленда и он заменился на снег, теперь мы убираем снег и получаем.. плейтинг, хотя изначально там был стеклянный пол. Можно ли сделать так чтобы эти лавалендовские турфы сохраняли что за пол там был раньше?

Хм. Это не совсем легко, потому что нужно изменять способы замены в самих турфах. Стоит делать ради такого? можно конечно и попробовать...

Dimach commented:
Для снега хотелось бы этот фунционал, для остального можно не делать

Rerik007 commented:
Я могу сделать замену на свой эффект в кристалле, вместо замены полов, также как после снег машины. Поможет ли это?

Dimach commented:
Снег после снегмашины имеет недостаток - он тает

Rerik007 commented:
Да, но там можно легко убрать это. С новым типом. Впрочем могу попробовать изменить и турфы снега..
Upd. Решил турфы возвращать.

Rerik007 committed:
if you changed turf - old returned

Подходит для тем джунглей и зимы. Лаваленд не стал менять, потому что бы накладываем новые плитки. Впрочем можно сделав другой способ убирания сделать. alien скин создаёт свои платинги(которые впрочем со сломанной иконкой и я не уверен что нужны), не стал менять.
